src/d3a/d3a_core/area_serializer.py

```diff
@@ -135,8 +135,9 @@ def optional(attr):
             children = [area_from_dict(child) for child in description['children']]
         else:
             children = None
+        grid_fee_percentage = description.get('grid_fee_percentage', None)
```

Comment:
I would call the `validate_area` method here instead of the constructor. That way we are going to keep the constructor of the class smaller, and the Area object will not even be created.

Reply:
After offline discussion, leaving it as it is.